### ISSUE TITLE
SugarTheme fatal errors

### ISSUE THREAD

**Issue by adamjakab** [salesagility/SuiteCRM#1718]
I have thousands of these in apache log each day:
`mod_fcgid: stderr: PHP Fatal error:  Call to a member function warn() on a non-object in /home/crm/httpdocs/include/SugarTheme/SugarTheme.php on line 911`
#### Possible Fix

a) create static functions in LoggerManager and call them directly and remove insane `$GLOBALS['log']->something();` calls from application.

b) put isset checks around usage of all $GLOBALS['log'] calls
#### Context

It seems not to be affecting the application (not sure) the 2 referrers I see are:
`index.php?action=index&amp;module=Calendar&amp;view=shared&amp;&amp;year=2016&amp;month=7&amp;day=10&amp;hour=0
index.php?module=Cases&amp;action=DetailView&amp;record=4d60933a-515e-d8c2-e78c-577f8a283723`
#### Your Environment

SuiteCRM 7.6.4
Sugar  6.5.23 (Build 1061)

**Comment by gunnicom** [comment]
+1 for a)

**Comment by pgorod** [comment]
I am all for cleaning up useless messages in logs. We are all black-belts in `grep`ing by now, but the truth is the day always comes when something important scrolls past you because of excessive log garbage, and you waste your valuable time and your limited patience...

**Comment by adamjakab** [comment]
@pgorod * I have actualy given up looking the suitecrm.log file filled with useless info.
A line like this: 
`Mon Jul 11 09:29:10 2016 [15192][-none-][FATAL]  Query Failed: SELECT count(*) c FROM ( FROM accounts  WHERE accounts.deleted = 0 ) as n: MySQL error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'FROM accounts  WHERE accounts.deleted = 0 ) as n' at line 1`
is absolutely useless if I cannot figure out from the logs in which file it occurred and what is the backtraced call stack to that point.

Fixing the LoggerManager could help us with that a lot.

Also, stuff like [this](https://github.com/salesagility/SuiteCRM/blob/master/include/database/MysqliManager.php#L131):  
`$GLOBALS['log']->info('Query:' . $sql);` should if at all be at 'debug' level - but I'd remove it alltogether.

**Comment by shogunpol** [comment]
@adamjakab , can you give the more details about how to replicate this bug,
for example the url of pages, and so on.

**Comment by pgorod** [comment]
@shogunpol since Adam is not answering, I would say that if you simply grep your Apache logs for Fatal errors you should see plenty of those. Then look up the line of code mentioned and you should find one of those `GLOBALS['log']` commands.

This is more of a generic problem with the way we're doing logging than a specific bug.

**Comment by adamjakab** [comment]
@shogunpol - hi - sorry foor the delay.
I put the referrals in the original post - not very clear but there - in the apache log I do not have other than referrals because it is php throwing the errors. However, as far as I can understand, I do not see any pages with fatal error messages in browser so i guess these must be ajax requests failing silently.

here are my last 20 fatal errors:

```
[Fri Aug 12 15:19:41.500125 2016] [fcgid:warn] [pid 31974] [client 79.58.159.xxx:63397] mod_fcgid: stderr: PHP Fatal error:  Call to a member function warn() on a non-object in /home/xxx/httpdocs/include/SugarTheme/SugarTheme.php on line 911, referer: http://xxx/index.php?module=Accounts&amp;action=index
[Fri Aug 12 15:19:46.485466 2016] [fcgid:warn] [pid 32315] [client 79.58.159.xxx:61662] mod_fcgid: stderr: PHP Fatal error:  Call to a member function warn() on a non-object in /home/xxx/httpdocs/include/SugarTheme/SugarTheme.php on line 911, referer: http://xxx/index.php?module=Accounts&amp;offset=1&amp;stamp=1471015178033990700&amp;return_module=Accounts&amp;action=DetailView&amp;record=3686e5b7-0199-e8f6-7d1a-56026c9fefab
[Fri Aug 12 15:20:25.755458 2016] [fcgid:warn] [pid 32444] [client 79.58.159.xxx:57626] mod_fcgid: stderr: PHP Fatal error:  Call to a member function warn() on a non-object in /home/xxx/httpdocs/include/SugarTheme/SugarTheme.php on line 911, referer: http://xxx/index.php?action=index&amp;module=Calendar&amp;view=shared&amp;&amp;year=2016&amp;month=8&amp;day=21&amp;hour=0
[Fri Aug 12 15:20:38.761974 2016] [fcgid:warn] [pid 32662] [client 79.58.159.xxx:64344] mod_fcgid: stderr: PHP Fatal error:  Call to a member function warn() on a non-object in /home/xxx/httpdocs/include/SugarTheme/SugarTheme.php on line 911, referer: http://xxx/index.php?module=Cases&amp;action=index
[Fri Aug 12 15:20:46.646760 2016] [fcgid:warn] [pid 32431] [client 79.58.159.xxx:58824] mod_fcgid: stderr: PHP Fatal error:  Call to a member function warn() on a non-object in /home/xxx/httpdocs/include/SugarTheme/SugarTheme.php on line 911, referer: http://xxx/index.php?module=Accounts&amp;offset=1&amp;stamp=1471015178033990700&amp;return_module=Accounts&amp;action=DetailView&amp;record=3686e5b7-0199-e8f6-7d1a-56026c9fefab
```

apart from this info id say follow as @pgorod advised

**Comment by samus-aran** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [SugarTheme fatal errors](https://trello.com/c/dGzptinF/87-sugartheme-fatal-errors)

**Comment by samus-aran** [comment]
We have now set up a new home for suggestions at Trello.  All github issues that were labeled 'suggestion' have been moved and will be closed.  Certain ones will be progressed within the new Suggestion Box and may be re-opened.

Announcement of moving Suggestions:
https://suitecrm.com/forum/suggestion-box/13691-moving-suggestions-from-github-to

New SuiteCRM Suggestion Box
https://trello.com/b/Ht7LbMqw/suitecrm-suggestion-box
